xwiki-platform-core/xwiki-platform-store/xwiki-platform-store-filesystem-oldcore/src/main/java/org/xwiki/store/filesystem/internal/DefaultTemporaryAttachmentSessionsManager.java

```diff
@@ -97,7 +104,13 @@ public XWikiAttachment uploadAttachment(DocumentReference documentReference, Par
         XWikiContext context = this.contextProvider.get();
         try {
             xWikiAttachment = new XWikiAttachment();
-            xWikiAttachment.setFilename(part.getSubmittedFileName());
+            String actualFilename;
+            if (filename != null) {
```

Comment:
Shouldn't we check for blank string too here?

Reply:
Indeed, could be safer.

Reply:
Done https://github.com/xwiki/xwiki-platform/pull/1930/files#diff-f8de10bbbc165fc4673c7bf82e7240394b5f62e332c43d906c5a4536652591c5R109